File: pdv/src/types/sale.ts
export type PaymentMethod = 'cash' | 'card_debit' | 'card_credit' | 'pix' | 'voucher'
export type SaleStatus    = 'completed' | 'cancelled' | 'pending'

export interface CartItem {
  productUuid:     string
  variantUuid?:    string | null  // variante selecionada (moda: tamanho/cor)
  productName:     string
  productSku:      string
  attributesJson?: string         // JSON snapshot de atributos da variante
  cost?:           number         // custo em centavos (para snapshot histórico)
  quantity:        number
  unitPrice:       number  // centavos
  discount:        number  // centavos
  total:           number  // centavos
}

export interface CartPayment {
  method:    PaymentMethod
  amount:    number
  reference: string
}

export interface SaleItem {
  uuid:                 string
  sale_uuid:            string
  product_uuid:         string
  variant_uuid:         string | null
  product_name:         string
  product_sku:          string
  name_snapshot:        string
  sku_snapshot:         string
  attributes_snapshot:  string
  cost_snapshot:        number | null
  quantity:             number
  unit_price:           number
  discount:             number
  total:                number
}

export interface Payment {
  uuid:      string
  sale_uuid: string
  method:    string
  amount:    number
  reference: string | null
}

export interface Sale {
  uuid:          string
  tenant_uuid:   string
  store_uuid:    string
  customer_uuid: string | null
  customer_name: string | null
  user_uuid:     string
  status:        SaleStatus
  subtotal:      number
  discount:      number
  total:         number
  notes:         string | null
  completed_at:  string | null
  synced_at:     string | null
  created_at:    string
}

export interface SaleDetail {
  sale:     Sale
  items:    SaleItem[]
  payments: Payment[]
}

export interface CreateSaleItemInput {
  product_uuid:          string
  variant_uuid?:         string | null
  product_name:          string
  product_sku:           string
  name_snapshot?:        string
  sku_snapshot?:         string
  attributes_snapshot?:  string
  cost_snapshot?:        number | null
  quantity:              number
  unit_price:            number
  discount?:             number
}

export interface CreatePaymentInput {
  method:     string
  amount:     number
  reference?: string
}

export interface CreateSaleInput {
  customer_uuid?: string
  user_uuid:      string
  discount?:      number
  notes?:         string
  items:          CreateSaleItemInput[]
  payments:       CreatePaymentInput[]
}

File: pdv/src-tauri/src/database/models/sale.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Sale {
    pub uuid:          String,
    pub tenant_uuid:   String,
    pub store_uuid:    String,
    pub customer_uuid: Option<String>,
    pub customer_name: Option<String>,
    pub user_uuid:     String,
    pub status:        String,
    pub subtotal:      i64,
    pub discount:      i64,
    pub total:         i64,
    pub notes:         Option<String>,
    pub completed_at:  Option<String>,
    pub synced_at:     Option<String>,
    pub created_at:    String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SaleItem {
    pub uuid:                 String,
    pub sale_uuid:            String,
    pub product_uuid:         String,
    pub variant_uuid:         Option<String>,
    pub product_name:         String,
    pub product_sku:          String,
    pub name_snapshot:        String,
    pub sku_snapshot:         String,
    pub attributes_snapshot:  String,
    pub cost_snapshot:        Option<i64>,
    pub quantity:             i64,
    pub unit_price:           i64,
    pub discount:             i64,
    pub total:                i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Payment {
    pub uuid:      String,
    pub sale_uuid: String,
    pub method:    String,
    pub amount:    i64,
    pub reference: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct SaleDetail {
    pub sale:     Sale,
    pub items:    Vec<SaleItem>,
    pub payments: Vec<Payment>,
}

#[derive(Debug, Deserialize)]
pub struct CreateSaleInput {
    pub customer_uuid: Option<String>,
    pub user_uuid:     String,
    pub discount:      Option<i64>,
    pub notes:         Option<String>,
    pub items:         Vec<CreateSaleItemInput>,
    pub payments:      Vec<CreatePaymentInput>,
}

#[derive(Debug, Deserialize)]
pub struct CreateSaleItemInput {
    pub product_uuid:         String,
    pub variant_uuid:         Option<String>,
    pub product_name:         String,
    pub product_sku:          String,
    pub name_snapshot:        Option<String>,
    pub sku_snapshot:         Option<String>,
    pub attributes_snapshot:  Option<String>,
    pub cost_snapshot:        Option<i64>,
    pub quantity:             i64,
    pub unit_price:           i64,
    pub discount:             Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct CreatePaymentInput {
    pub method:    String,
    pub amount:    i64,
    pub reference: Option<String>,
}
